### PR TITLE
Remove caching winning option data and bit trick for formula params

### PR DESCRIPTION
**Pull request by manhlx3006**
The gas cost reduces around 3k-5k for the first trade of each epoch after removing the caching

### CODE REVIEW

File: contractsSol5/stakingAndDao/KyberDAO.sol [hunk 1]
```diff
@@ -117,9 +116,6 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
     // campOptionPoints[campID]: total points and points of each option for a campaign
```
Comment (ilanDoron) [inline contractsSol5/stakingAndDao/KyberDAO.sol]: reading this again.
for me total votes
is much more clear then total points.

I am not native English.
maybe consult someone else as well.

File: contractsSol5/stakingAndDao/KyberDAO.sol [hunk 2]
```diff
@@ -102,10 +101,10 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
         uint campID;
         uint startBlock;
         uint endBlock;
-        uint totalKNCSupply;    // total KNC supply at the time campaign was created
-        uint formulaParams;     // squeezing formula params into one number
-        bytes link;             // link to KIP, explaination of options, etc.
-        uint[] options;         // data of options
+        uint totalKNCSupply;        // total KNC supply at the time campaign was created
+        FormulaData formulaData;  // formula params for concluding campaign result
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply contractsSol5/stakingAndDao/KyberDAO.sol]: Updated

File: contractsSol5/stakingAndDao/KyberDAO.sol [hunk 3]
```diff
@@ -209,7 +205,8 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
 
     event NewCampaignCreated(
         CampaignType campType, uint campID,
-        uint startBlock, uint endBlock, uint formulaParams,
+        uint startBlock, uint endBlock,
+        uint minPerInPrecision, uint cInPrecision, uint tInPrecision,
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply contractsSol5/stakingAndDao/KyberDAO.sol]: Ok I changed to minPercentageInPercision

File: contractsSol5/stakingAndDao/KyberDAO.sol [hunk 4]
```diff
@@ -536,7 +537,7 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
         // more than 1 options have same vote count
         if (winningOption == 0) { return (0, 0); }
 
-        FormulaData memory formulaData = decodeFormulaParams(camp.formulaParams);
+        FormulaData memory formulaData = camp.formulaData;
```
[the remark this answers is not in the document]
Comment (manhlx3006) [reply contractsSol5/stakingAndDao/KyberDAO.sol]: From my view is this data only write once when submitting new campaign, and get once for the first trad. So I don't think it is needed to compress :/

**Pull request merged**